src/Handler.cpp

```diff
@@ -802,6 +802,15 @@ void Handler::handleResponse(ConnectConnection* s, Request* req, Response* res)
                 c->peer(), c->fd(), c->status(), c->statusStr());
         return;
     }
+    if (res->type() == Reply::Error) {
+        if (res->isLoading()) {
+            Server* serv = s->server();
+            serv->setFail(true);
+            logNotice("server %s mark failure because it is loading error", serv->addr().data());
+            handleRequest(req);
```

Comment:
can potentially go in infinite loop. Should there be an upper cap on retries?

Reply:
Yes, will add this.

Reply:
Found another bug, will fix both in the next commit.

Reply:
fixed in a11d93f, testing this more with integration and unit